src/framework/parsers/draco-worker.js

```diff
@@ -0,0 +1,259 @@
+        switch (dataType) {
+            case draco.DT_UINT8: return new Uint8Array(typedArray.buffer, typedArray.byteOffset, typedArray.byteLength);
+            case draco.DT_UINT16: return new Uint16Array(typedArray.buffer, typedArray.byteOffset, typedArray.byteLength / 2);
+            case draco.DT_UINT32: return new Uint32Array(typedArray.buffer, typedArray.byteOffset, typedArray.byteLength / 4);
+            case draco.DT_FLOAT32: return new Float32Array(typedArray.buffer, typedArray.byteOffset, typedArray.byteLength / 4);
+        }
+        return null;
+    }
+
+    const componentSizeInBytes = (dataType) => {
+        switch (dataType) {
+            case draco.DT_UINT8: return 1;
+            case draco.DT_UINT16: return 2;
+            case draco.DT_UINT32: return 4;
+            case draco.DT_FLOAT32: return 4;
+        }
+        return 1;
+    };
+
+    const attributeSizeInBytes = (attribute) => {
+        return attribute.num_components() * componentSizeInBytes(attribute.data_type());
+    };
+
+    const attributeOrder = {
+        0: 0,   // position
+        1: 1,   // normal
+        5: 2,   // tangent
+        2: 3,   // color
+        7: 4,   // joints
+        8: 5,   // weights
+        4: 6,   // generic (used for blend indices and weights)
+        3: 7,   // texcoord
+    };
+
+    const generateNormals = (vertices, indices) => {
+        const subtract = (dst, a, b) => {
+            dst[0] = a[0] - b[0];
+            dst[1] = a[1] - b[1];
+            dst[2] = a[2] - b[2];
+        };
+    
+        const cross = (dst, a, b) => {
+            dst[0] = a[1] * b[2] - b[1] * a[2];
+            dst[1] = a[2] * b[0] - b[2] * a[0];
+            dst[2] = a[0] * b[1] - b[0] * a[1];
+        };
+    
+        const normalize = (dst, offset) => {
+            const a = dst[offset + 0];
+            const b = dst[offset + 1];
+            const c = dst[offset + 2];
+            const l = 1.0 / Math.sqrt(a * a + b * b + c * c);
+            dst[offset + 0] *= l;
+            dst[offset + 1] *= l;
+            dst[offset + 2] *= l;
+        };
+    
+        const copy = (dst, src, srcOffset) => {
+            for (let i = 0; i < 3; ++i) {
+                dst[i] = src[srcOffset + i];
+            }
+        };
+    
+        const numTriangles = indices.length / 3;
+        const numVertices = vertices.length / 3;
+        const result = new Float32Array(vertices.length);
+        const a = [0, 0, 0],
+              b = [0, 0, 0],
+              c = [0, 0, 0],
+              t1 = [0, 0, 0],
+              t2 = [0, 0, 0],
+              n = [0, 0, 0];
+
+        for (let i = 0; i < numTriangles; ++i) {
+            const v0 = indices[i * 3 + 0] * 3;
+            const v1 = indices[i * 3 + 1] * 3;
+            const v2 = indices[i * 3 + 2] * 3;
+
+            copy(a, vertices, v0);
+            copy(b, vertices, v1);
+            copy(c, vertices, v2);
+
+            subtract(t1, b, a);
```

Comment:
are positions here always in a format we can use math operations on?

Reply:
yep, this logic hasn't changed.